src/Core/Models/Data/SsoConfigurationData.cs

```diff
@@ -13,11 +14,25 @@ public class SsoConfigurationData
         private const string _oidcSignedOutPath = "/oidc-signedout";
         private const string _saml2ModulePath = "/saml2";
 
+        private static JsonSerializerOptions _jsonSerializerOptions = new()
+        {
+            PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+        };
+
+        public static SsoConfigurationData Deserialize(string data)
+        {
+            return JsonSerializer.Deserialize<SsoConfigurationData>(data, _jsonSerializerOptions);
+        }
+
+        public string Serialize()
+        {
+            return JsonSerializer.Serialize(this, _jsonSerializerOptions);
+        }
```

Comment:
We have a generic method that accomplishes this exact thing in CoreHelpers. It may still be useful to have hardcoded type information on each specific class that has serialized data, but those methods should at least call the global generic method instead of duplicating the defaults and calls to the serializer.